src/main/java/janggi/domain/Piece.java

```diff
@@ -0,0 +1,55 @@
+package janggi.domain;
+
+import janggi.domain.piece.Cannon;
+import janggi.domain.piece.General;
+import janggi.domain.piece.PieceBehavior;
+import java.util.Objects;
+import java.util.Set;
+
+public class Piece {
+
+    private final Side side;
+    private final PieceBehavior pieceBehavior;
+
+    public Piece(Side side, PieceBehavior pieceBehavior) {
+        this.side = side;
+        this.pieceBehavior = pieceBehavior;
+    }
+
+    public boolean isSameSide(Side side) {
+        return this.side == side;
+    }
+
+    public boolean isCannon() {
+        return pieceBehavior instanceof Cannon;
```

Comment:
instanceof 를 제거하고 아래와 같은 형태로 변경해볼까요~?

```suggestion
        return pieceBehavior.isCannon();
```

Reply:
우선은 인터페이스에 isCannon을 추가해서 처리했어요. 우선은 ISP를 위반하는 것이라고 신경이 쓰이고 있습니다 ㅠㅠㅠ

내일 좀 더 고민해보도록 하겠습니다!